src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -45,38 +41,37 @@ internal void WriteAsV7(Stream archiveStream, Span<byte> buffer)
         }
 
         // Asynchronously writes the current header as a V7 entry into the archive stream and returns the value of the final checksum.
-        internal async Task<int> WriteAsV7Async(Stream archiveStream, Memory<byte> buffer, CancellationToken cancellationToken)
+        internal async Task WriteAsV7Async(Stream archiveStream, Memory<byte> buffer, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            long actualLength = GetTotalDataBytesToWrite();
-            TarEntryType actualEntryType = TarHelpers.GetCorrectTypeFlagForFormat(TarEntryFormat.V7, _typeFlag);
-
-            int tmpChecksum = WriteName(buffer.Span, out _);
-            tmpChecksum += WriteCommonFields(buffer.Span, actualLength, actualEntryType);
-            int checksum = WriteChecksum(tmpChecksum, buffer.Span);
+            long actualLength = WriteV7FieldsToBuffer(buffer.Span);
 
             await archiveStream.WriteAsync(buffer, cancellationToken).ConfigureAwait(false);
 
             if (_dataStream != null)
             {
                 await WriteDataAsync(archiveStream, _dataStream, actualLength, cancellationToken).ConfigureAwait(false);
             }
+        }
 
-            return checksum;
+        // Writes the V7 header fields to the specified buffer, calculates and writes the checksum, then returns the final data length.
+        private long WriteV7FieldsToBuffer(Span<byte> buffer)
+        {
+            long actualLength = GetTotalDataBytesToWrite();
+            TarEntryType actualEntryType = TarHelpers.GetCorrectTypeFlagForFormat(TarEntryFormat.V7, _typeFlag);
+
+            int tmpChecksum = WriteName(buffer, out _);
+            tmpChecksum += WriteCommonFields(buffer, actualLength, actualEntryType);
+            _checksum = WriteChecksum(tmpChecksum, buffer);
```

Comment:
Now I can store the checksum directly in the current instance (this created a bug in a previous PR).